libmachine/provision/utils.go

```diff
@@ -172,10 +174,17 @@ func ConfigureAuth(p Provisioner) error {
 		return err
 	}
 
-	// TODO: Do not hardcode daemon port, ask the driver
-	if err := utils.WaitForDocker(ip, dockerPort); err != nil {
-		return err
-	}
+	waitForDocker(p, dockerPort)
 
 	return nil
 }
+
+func waitForDocker(p Provisioner, dockerPort int) {
+	utils.WaitForSpecific(func() bool {
+		// HACK: Check netstat's output to see if anyone's listening on the Docker API port.
+		if _, err := p.SSHCommand(fmt.Sprintf("netstat -a | grep %d", dockerPort)); err != nil {
```

Comment:
Maybe something like `netstat -ln | grep ':::2376'`?  That starts to get into kind of specific output though, and I'm not sure how much the output of `netstat` deviates between systems.